frontend/src/views/admin/matching/index.tsx

```diff
@@ -21,12 +68,329 @@ export function AdminMatchingView() {
         }
     }, [activeSession, dispatch]);
 
+    // Compute and keep track of the combined applications for each applicant
+    const combinedApplicationsByApplicantId: Record<number, Application> =
+        React.useMemo(() => {
+            const ret: Record<number, Application> = {};
+            for (const applicant of applicants) {
+                const combinedApplication = getCombinedApplication(
+                    applicant,
+                    applications
+                );
+                if (combinedApplication) {
+                    ret[applicant.id] = combinedApplication;
+                }
+            }
+
+            return ret;
+        }, [applicants, applications]);
+
+    // Initialize the set of matches in Redux on loading the component:
+    React.useEffect(() => {
```

Comment:
Mm... I'm trying to initially populate `state.ui.matchingData.matches` with matches based on what applicants have applied to and what they've been assigned to.

I realize now that this is probably what's causing issues when I tried to implement local storage before as well, since it's re-initializing the state of `matchingData.matches` every time you reload/tab into the Matching component.

Maybe a more correct way would be to only create a `Match` when an applicant's status is something _other_ than `applied`. PositionSummary could have a list of applicants and assignments, as well as a separate list of matches.

The `Match` type could also be modified then to have `assigned`, `starred` and `hidden` as separate boolean fields to also address another previous comment?

Reply:
Updated code in an attempt to make changes discussed here and in the meeting. 
Leaving this conversation as unresolved, but please add a reply or mark as resolved after review. :)